src/openms/source/ANALYSIS/OPENSWATH/DIAHelper.cpp

```diff
@@ -273,29 +274,30 @@ namespace OpenMS
     } //end of dia_isotope_corr_sub
 
     //simulate spectrum from AASequence
-    void simulateSpectrumFromAASequence(const AASequence& aa,
+    void simulateSpectrumFromAASequence(IsotopeDistributionCache& iso,
+                                        const AASequence& aa,
                                         std::vector<double>& firstIsotopeMasses, //[out]
                                         std::vector<std::pair<double, double> >& isotopeMasses, //[out]
                                         TheoreticalSpectrumGenerator const * generator, double charge)
     {
       getTheorMasses(aa, firstIsotopeMasses, generator, charge);
       for (std::size_t i = 0; i < firstIsotopeMasses.size(); ++i)
       {
-        getAveragineIsotopeDistribution(firstIsotopeMasses[i], isotopeMasses,
+        getAveragineIsotopeDistribution(iso, firstIsotopeMasses[i], isotopeMasses,
                                         charge);
       }
     }
 
     //given an experimental spectrum add isotope pattern.
-    void addIsotopes2Spec(const std::vector<std::pair<double, double> >& spec,
+    void addIsotopes2Spec(IsotopeDistributionCache& iso, const std::vector<std::pair<double, double> >& spec,
                           std::vector<std::pair<double, double> >& isotopeMasses, //[out]
                           double charge)
     {
 
       for (std::size_t i = 0; i < spec.size(); ++i)
       {
         std::vector<std::pair<double, double> > isotopes;
```

Comment:
just as a suggestion for speed (even though this was not part of the PR):
move `std::vector<std::pair<double, double> > isotopes;` in front of the loop and just clear it inside to safe a lot of allocations.